apps/pipelines/tests/test_runnable_builder.py

```diff
@@ -160,6 +170,43 @@ def test_llm_response(get_llm_service, provider, pipeline):
     assert runnable.invoke(PipelineState(messages=["Repeat exactly: 123"]))["messages"][-1] == "123"
 
 
+@django_db_with_data(available_apps=("apps.service_providers",))
+@mock.patch("apps.service_providers.models.LlmProvider.get_llm_service")
+@mock.patch("apps.pipelines.nodes.base.PipelineNode.logger", mock.Mock())
+def test_llm_with_prompt_response(get_llm_service, provider, pipeline, source_material, experiment_session):
+    service = build_fake_llm_service(responses=["123"], token_counts=[0])
+    get_llm_service.return_value = service
+    data = {
+        "edges": [],
+        "nodes": [
+            {
+                "data": {
+                    "id": "llm-GUk0C",
+                    "label": "Get the robot to respond",
+                    "type": "LLMResponseWithPrompt",
+                    "params": {
+                        "llm_provider_id": provider.id,
+                        "llm_model": "fake-model",
+                        "source_material_id": source_material.id,
+                        "prompt": "Use this {source_material} to answer questions about {participant_data}."
```

Comment:
shouldn't this also include the `{input}` var?

Reply:
I updated the test to have multiple nodes, which also includes that variable. https://github.com/dimagi/open-chat-studio/pull/608/commits/80b35e59a1e4ec4b8145151402bf2571f9a96d2a